### ISSUE TITLE
Set custom fields when editing multiple transactions

### ISSUE THREAD

**Issue by cmbarbu** [moneymanagerex/moneymanagerex#4207]
# Description of the feature

I recently (re)discovered MMEX's custom fields manager. It is a very useful feature that allows me to enrich transactions with additional information (e.g., loyalty points from merchants). However, if I want to update many past transactions with the same custom field value, I have to edit them one by one. Sadly, the excellent Multi Transactions Update does not yet support custom fields.

Therefore, the feature request is to **allow users to set/unset/update custom fields when using the multiple transactions feature.**

Related to https://github.com/moneymanagerex/moneymanagerex/issues/4159

![mmex-feature-custom-fields](https://user-images.githubusercontent.com/1184162/154363017-cdc0123e-a38f-466f-bdac-98ccaa3e288e.PNG)

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/154532636-5e9da819-e589-4887-a578-c69b44f4e673.png)

**Comment by tactilis** [comment]
@vomikan 

The field widths have expanded for some reason.

Single transaction edit and multi-transaction edit shown at same scale.

![image](https://user-images.githubusercontent.com/4060092/154562701-fdfd4fe5-2670-46e8-9762-43143e66d9e1.png)

**Comment by vomikan** [comment]
@tactilis I can reproduce this issue. It happens because some payee mane lenght is too long.
The same will happen if some [Transfer to] account name  is too long.

**Comment by tactilis** [comment]
You are correct. I have a long payee name that came from testing somebody's CSV statement import:

![image](https://user-images.githubusercontent.com/4060092/154567745-12f8ae55-cbd7-4f21-904a-82c110c07b27.png)

But the long name doesn't make the single transaction edit dialog become wider than normal.

The field widths on the single transaction edit dialog are constrained. I believe the multi transaction edit dialog should have its field widths similarly constrained.

**Comment by tactilis** [comment]
When editing multiple transactions, you also need an additional 'Specify' tick box, like on the left hand side of the dialog but in the custom fields panel, for the user to indicate whether they are editing the custom fields or not.

Consider this:

There are 3 transactions selected for editing with custom fields values of XX, YY, and [not set].

![image](https://user-images.githubusercontent.com/4060092/154566148-515f188d-6702-4bd9-bb69-6cffc9ed912d.png)

I multi-edit the transactions and open the custom fields panel. Observe that the custom field is not ticked.

![image](https://user-images.githubusercontent.com/4060092/154566741-203fd8e6-6da1-436f-9900-b93174553d58.png)

Without changing anything, I click OK and find that the transactions previously having the XX and YY custom fields are now [not set].

![image](https://user-images.githubusercontent.com/4060092/154567113-0524973f-a236-47d9-963a-30ddfc50655c.png)

By default the new 'Specify' tick box in the custom fields panel should be unticked, like the tick boxes on the left.

**Comment by tactilis** [comment]
Thinking about this further, I believe that there should be a separate Specify tick box for each custom field type.

Originally I was thinking just one tick box for all custom fields, but that is not sufficient.

**Comment by vomikan** [comment]
@tactilis I don't understand your explanation of the problem.

**Comment by tactilis** [comment]
I think the screenshots explain it exactly.

Try setting up the scenario I have shown:

I have one custom field Trans Type, with three state XX, YY, ZZ

![image](https://user-images.githubusercontent.com/4060092/154571550-6ee5f1e6-f2f6-4b53-ac0c-da95b55a541e.png)


Apply XX to one transaction as shown.
Apply YY to the next transaction.
Don't set the custom field on the next transaction.

Now multi-edit all three transactions.  Just click OK (you don't even have to open the custom panel)

Now observe that XX and YY are gone from the first 2 transactions!

This is because the Trans Type tick box is part of the custom field state.

You need a 'Select' tick box for each custom field tick box & value to indicate whether that custom field is to be changed or not

**Comment by vomikan** [comment]
I think I've fixed the bug. Please check.

**Comment by tactilis** [comment]
Tested. All working as it should now.

I misunderstood the intended operation of the existing custom field tick boxes, so everything I said about needing an extra 'Select' tick box was nonsense!

This is a great addition to MMEX. Thanks Nikolay.

@cmbarbu will be happy too :grinning:

**Comment by cmbarbu** [comment]
Amazing, thank you for the quick implementation and bugfix!
